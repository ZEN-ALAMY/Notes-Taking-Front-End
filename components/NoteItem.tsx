function NoteItem(props) {
  return (
    <div className="flex xl:flex-row sm:flex-col flex-col bg-gray-300 xl:h-[70px] sm:h-[120px] h-120px my-[30px] xl:w-[95%] sm:w-[70%] w-[80%] mx-auto rounded-xl xl:items-center sm:justify-evenly justify-evenlyr">
      <div className="xl:w-[33.33%] ml-[5px]">
        TITLE:&nbsp;&nbsp;{props.title}
      </div>
      <div className="sm:text-center xl:w-[33.33%] xl:ml-0 sm:ml-0 ml-[5px]">
        DATE:&nbsp;&nbsp;{props.date}
      </div>
      <div className="sm:text-right xl:w-[33.33%] xl:ml-0 sm:ml-0 ml-[5px]">
        RELEVANCE:&nbsp;&nbsp;{props.relevance}
      </div>
    </div>
  );
}

export default NoteItem;
