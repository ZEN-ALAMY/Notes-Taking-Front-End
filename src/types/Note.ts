interface Note {
  title: string;
  date: string;
  relevance: string;
}

export type { Note };
