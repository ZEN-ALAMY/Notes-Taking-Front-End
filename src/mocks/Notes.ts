import { Note } from "../types/Note";

const MOCK_NOTE: Note = {
  title: "My Notes",
  date: "19-09-2022",
  relevance: "Not relevant",
};

const MOCK_NOTES = [MOCK_NOTE];

export { MOCK_NOTE, MOCK_NOTES };
