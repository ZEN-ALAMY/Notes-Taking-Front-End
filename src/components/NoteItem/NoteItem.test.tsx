import { render } from "@testing-library/react";
import { MOCK_NOTE } from "../../mocks/Notes";
import { Note } from "../../types/Note";
import NoteItem from "./NoteItem";

describe("NoteItem component tests", () => {
  it("should render the UI correctly", () => {
    expect(render(<NoteItem {...MOCK_NOTE} />).asFragment()).toMatchSnapshot();
  });
});
