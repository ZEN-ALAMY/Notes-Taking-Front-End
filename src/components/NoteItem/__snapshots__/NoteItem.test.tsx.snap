// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteItem component tests should render the UI correctly 1`] = `
<DocumentFragment>
  <div
    class="flex xl:flex-row sm:flex-col flex-col bg-gray-300 xl:h-[70px] sm:h-[120px] h-120px my-[30px] xl:w-[95%] sm:w-[70%] w-[80%] mx-auto rounded-xl xl:items-center sm:justify-evenly justify-evenlyr"
  >
    <div
      class="xl:w-[33.33%] ml-[5px]"
    >
      TITLE:  My Notes
    </div>
    <div
      class="sm:text-center xl:w-[33.33%] xl:ml-0 sm:ml-0 ml-[5px]"
    >
      DATE:  19-09-2022
    </div>
    <div
      class="sm:text-right xl:w-[33.33%] xl:ml-0 sm:ml-0 ml-[5px]"
    >
      RELEVANCE:  Not relevant
    </div>
  </div>
</DocumentFragment>
`;
